.github/workflows/status.yaml

```diff
@@ -0,0 +1,64 @@
+name: Custom GH Status from Workflow Artifacts
+
+on:
+  workflow_run:
+    workflows: ["pr"]
+    types:
+      - completed
+
+jobs:
+  process_artifacts:
+    if: ${{ github.event.workflow_run.conclusion == 'success' }}
+    runs-on: ubuntu-latest
+    steps:
+      - name: Download artifact
+        uses: actions/download-artifact@v4
+        with:
+          name: gh-status
+          path: downloaded_artifacts/
+          run-id: ${{ github.event.workflow_run.id }}
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+      - name: Read artifact JSON
+        id: read_artifact
+        run: |
+          CONTENT=$(cat downloaded_artifacts/gh-status.json)
+          echo "content=$CONTENT" >> $GITHUB_OUTPUT
+      - name: Create status
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          ARTIFACT_CONTENT: ${{ steps.read_artifact.outputs.content }}
+          COMMIT_SHA: ${{ github.event.workflow_run.head_sha }}
+          WORKFLOW_RUN_ID: ${{ github.event.workflow_run.id }}
+          REPO_NAME: ${{ github.repository }}
+          ATTEMPTS: ${{ github.event.workflow_run.run_attempt }}
+        run: |
+          JOB_ID=$(gh api repos/$REPO_NAME/actions/runs/$WORKFLOW_RUN_ID/jobs --paginate --jq '.jobs[] | select(.name == "pandas-tests-diff / build") | .id')
```

Comment:
Hardcoding `pandas-tests-diff` seems fragile. If the name of that job ever changes, this line would either fail or mess up the subsequent steps.

It looks like this value can be obtained from the `${{ github.job }}` property: https://docs.github.com/en/actions/learn-github-actions/contexts#github-context

![image](https://github.com/rapidsai/cudf/assets/7400326/d1da8b60-b205-44bc-a33a-5a1252afc3e3)

I would recommend adding that key/value pair to the `gh-status.json` artifact that gets uploaded and then consuming it here.

That will make this workflow more universal.

Reply:
Yup, made it more universal 👍